### PR TITLE
feat: Removing helper function.

### LINKED ISSUE #60: Update Plotly code for v1.0
The plotly code in the `dev` branch has been re-reviewed and contains some now redundant code. The changes needed prior to version 1.0 release are:
- [ ] Remove the helper functions in afcharts.py and theme_af.py in favour of the plotly template ([pio_template.py](https://github.com/best-practice-and-impact/afcharts-py/blob/dev/src/afcharts/assets/pio_template.py))
- [ ] Address PR #45 comments on the plotly template
- [ ] Add a basic usage example to the README
- [ ] Update cookbook examples accordingly

### PR DESCRIPTION
**Pull request by rodriguesfred**
## Description
Removing helper function.

### Summary
Refactoring cored to remove helper function functionality. I left the script there just in case.
Addressing @wfahim comments on other PR.
For the record assigning palette works a lot better and easier with helper function.

### Related Issues

- Fixes #
- Closes #60 
- Relates to #

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement
- [ ] 🔨 Build/CI changes
- [ ] 🎨 Style/formatting changes

## Changes Made

- 
- 
- 

### Test Coverage
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing performed
- [ ] No tests needed (explain why below)

### Testing Details
Ran existing pytests

**Test Cases:**
1. 
2. 
3. 

**Manual Testing Results:**

## Breaking Changes

- [ ] No breaking changes
- [ ] Breaking changes (describe below)

**Breaking Change Details:**

## Documentation
- [x] Documentation updated
- [x] README updated
- [ ] No documentation changes needed

**Documentation Changes:**
Updated cookbook and Readme.

## Dependencies
- [x] No new dependencies
- [ ] New dependencies added (list below)
- [ ] Dependencies updated (list below)
- [ ] Dependencies removed (list below)

**Dependency Changes:**

### Code Quality
- [x] Code follows the project's [style guidelines](https://3.basecamp.com/4322319/buckets/25844562/documents/8777425747)
- [x] Self-review of code completed
- [x] Code is properly commented
- [x] No unnecessary console.log/print statements
- [x] No commented-out code (unless intentionally documented)

### Testing & Quality Assurance
- [x] Tests pass locally
- [ ] Tests added for new functionality
- [ ] Edge cases considered and tested
- [ ] Error handling implemented where appropriate

### CODE REVIEW

File: src/afcharts/pio_template.py [hunk 1]
```diff
@@ -1,152 +1,155 @@
...
+        "hovermode": "x unified",  # How hovering affects the display
+        # x unified shows info for all the data at that point in the x-axis
+        "paper_bgcolor": "rgba(0,0,0,0)",  # Makes paper (entire area) background transparent
+        "plot_bgcolor": "rgba(0,0,0,0)",  # Makes plot area transparent
+        "margin": {  # Set margins around the plot area in pixels
+            "l": half_line,  # Left margin
+            "r": half_line,  # Right margin
+            "t": (base_size * 1.6) + (0.7 * (base_size * 1.6)) + half_line + base_size,
+            # Top margin is the size of the title + subtitle size + title_padding
+            # + space between title and subtitle (which is set by default)
```
Comment (nickhenden) [inline src/afcharts/pio_template.py:91]: As noted above, title, subtitle and source shouldn't be embedded in the chart anyway so we don't need to make allowances for these in the margins.

```suggestion
            "t": half_line,  # Top margin
```

**Review by nickhenden — CHANGES_REQUESTED** [review]
Thanks Fred. The only major change is to move the titles and subtitles into the text above the chart, rather than the chart itself. Might need some trial and error to work out the best way to achieve this. But hey, at least we don't have to worry about margins 🙃

**Comment by rodriguesfred** [comment]
@nickhenden have a look at the cookbook I think I did it.

**Comment by nickhenden** [comment]
> @nickhenden have a look at the cookbook I think I did it.

Nice one. Thanks @rodriguesfred I'll take a look :)

File: src/afcharts/cookbook/03-plotly-usage.qmd [hunk 2]
```diff
@@ -1,162 +1,187 @@
...
+<div class="chart-subtitle">Population of countries in the Americas (millions), 2007</div>
+
+```{python}
+#| ref-label: bar-charts-1
+#| echo: false
+#| eval: true
+
+make_bar_chart()
+
+```
```
Comment (nickhenden) [inline src/afcharts/cookbook/03-plotly-usage.qmd:49]: @rodriguesfred I tried to find a way to do this without creating a function because some users will be python newbies. There isn't a handy way to reuse chunks like there is for R, so I think the following is probably the best option.

It uses `output: false` on the first chunk to show the code but not the figure. Then the second chunk, which runs `fig.show()` again has  `echo: false` to show the figure but not the code.

P.S. I will do a proper review later.

```
## Bar charts

#```{python bar-charts-1}
#| output: false

import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

# AF Package
from afcharts.pio_template import pio

# Load the gapminder dataset from plotly.express
df = px.data.gapminder().query("year == 2007 & continent == 'Americas'")

top5 = df.nlargest(5, "pop")
fig = go.Figure()

fig.add_trace(
    go.Bar(
        x=top5["country"],
        y=top5["pop"],  # Repeat the category name for each x value
        name="test bar",  # Use the x value column name as the trace name
    )
)

# Update layout
fig.update_layout(
    height=300
)

fig.show()

#```

<div class="chart-title">The U.S.A. is the most populous country in the Americas</div>
<div class="chart-subtitle">Population of countries in the Americas (millions), 2007</div>

#```{python}
#| echo: false

fig.show()

#```
```

File: src/afcharts/cookbook/03-plotly-usage.qmd [hunk 3]
```diff
@@ -1,162 +1,187 @@
...
+                line=dict(color=duo[i % len(duo)])
+            )
+        )
 
-df = df[df["country"].isin(["United Kingdom", "China"])]
+    # Update layout
+    fig.update_layout(
+        xaxis=dict(
+            title="Year",
+            showgrid=False  # Hide x-axis grid lines
```
Comment (nickhenden) [inline src/afcharts/cookbook/03-plotly-usage.qmd:94]: Can we switch off x-axis grid lines by default in the template instead?

File: src/afcharts/cookbook/styles.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
+.chart-source {
+    font-size: 12pt;
+    font-weight: 700;
```
Comment (nickhenden) [inline src/afcharts/cookbook/styles.css:3]: I think put sources as regular text not bold. That's how they've done it in the [Analysis Function guidance](https://analysisfunction.civilservice.gov.uk/policy-store/data-visualisation-charts/).
```suggestion
```

**Review by nickhenden — CHANGES_REQUESTED** [review]
Thanks again @rodriguesfred . The main change is to remove the functions as described in my previous comment. Otherwise, it's all tiny changes.

File: src/afcharts/pio_template.py [hunk 5]
```diff
@@ -1,152 +1,153 @@
...
+                "standoff": half_line / 2,  # Position from axes
+            },
+            "fixedrange": True,
+            "zeroline": True,
+            "zerolinecolor": af_colour_values["chart_features"],
+        },
+    }
+)
+
+pio.templates.default = "theme_af"
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply src/afcharts/pio_template.py:153]: I changed my mind about this and decided to explicitly call the theme. I think its clearer on what is hapening and more flexible.
